### PR TITLE
Revert "Do not use subprocess in multipart download"

### PR DESCRIPTION
**Pull request by harupy**
Reverts mlflow/mlflow#8561

### CODE REVIEW

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

File: mlflow/store/artifact/databricks_artifact_repo.py [hunk 1]
```diff
@@ -60,7 +60,7 @@
 )
 
 _logger = logging.getLogger(__name__)
-_DOWNLOAD_CHUNK_SIZE = 10_000_000  # 10 MB
+_DOWNLOAD_CHUNK_SIZE = 100_000_000  # 100 MB
```
Comment (harupy) [inline mlflow/store/artifact/databricks_artifact_repo.py:63]:
```suggestion
_DOWNLOAD_CHUNK_SIZE = 500_000_000  # 500 MB
```

File: mlflow/store/artifact/databricks_models_artifact_repo.py [hunk 2]
```diff
@@ -23,6 +22,7 @@
 )
 
 _logger = logging.getLogger(__name__)
+_DOWNLOAD_CHUNK_SIZE = 100_000_000  # 100 MB
```
Comment (harupy) [inline mlflow/store/artifact/databricks_models_artifact_repo.py:25]:
```suggestion
_DOWNLOAD_CHUNK_SIZE = 500_000_000  # 500 MB
```

**Comment by harupy** [comment]
@dbczumar @BenWilson2 I reran python and windows checks because they failed with strange errors. Feel free to merge once they pass.

**Comment by harupy** [comment]
@dbczumar @BenWilson2 Feel free to merge this once CI finishes, or push commits if it fails.

**Pull request merged**